### PR TITLE
Move add_azure_core_repo function into operating_system.py

### PR DESCRIPTION
**Pull request by lubaihua33**
Move add_azure_core_repo function into operating_system.py, so this method can be used by other test suites.

### CODE REVIEW

File: lisa/operating_system.py [hunk 1]
```diff
@@ -763,6 +766,20 @@ def _get_package_information(self, package_name: str) -> VersionInfo:
         )
         return self._cache_and_return_version_info(package_name, version_info)
 
+    def add_azure_core_repo(self) -> None:
+        self.install_packages(["gnupg", "software-properties-common"])
+        # no azure-compatscanner package in azurecore-debian
+        # use azurecore instead
+        codename = "bionic"
```
Comment (LiliDeng) [inline lisa/operating_system.py:780]: please add comments for why hardcode codename here :)

File: lisa/operating_system.py [hunk 2]
```diff
@@ -1119,6 +1136,22 @@ def _get_information(self) -> OsInformation:
 
         return information
 
+    def add_azure_core_repo(self) -> None:
+        # there is no available repo for distro which higher than ubuntu 18.04,
+        # use bionic for temp solution
+        if self.information.version >= "18.4.0":
```
Comment (squirrelsc) [inline lisa/operating_system.py]: The code was written in 3 years ago. We should check other versions, and add support for 20.04, and >=22.04 as default.

File: lisa/operating_system.py (hunk 1 above)
Comment (lubaihua33) [reply lisa/operating_system.py:780]: Done

File: lisa/operating_system.py (hunk 2 above)
Comment (lubaihua33) [reply lisa/operating_system.py]: Done

**Comment by squirrelsc** [comment]
LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**